experimental/sql/dialects/rel_ssa.py

```diff
@@ -0,0 +1,320 @@
+    return Bag([ArrayAttr.from_list(schema_elts)])
+
+
+#===------------------------------------------------------------------------===#
+# Expressions
+#===------------------------------------------------------------------------===#
+
+
+class Expression(Operation):
+  """
+  Interface class for all Expressions, i.e., operations that work on single
+  values.
+  """
+  ...
+
+
+@irdl_op_definition
+class Column(Expression):
+  """
+  References a specific column with name `col_name` and returns its value. The
+  name 'col_name' must be part of the schema of an encompassing operator.
+
+  Example:
+
+  '''
+  %0 : rel_ssa.int32 = rel_ssa.column() ["col_name" = "id"]
+  '''
+  """
+  name = "rel_ssa.column"
+
+  col_name = AttributeDef(StringAttr)
+  result = ResultDef(DataType)
+
+
+@irdl_op_definition
+class Compare(Expression):
+  """
+  Returns `left` 'comparator' `right`.
+
+  Example:
+
+  '''
+  ...
+  %0 : !rel_ssa.bool = rel_ssa.compare(%a : !rel_ssa.int32, %b : !rel_ssa.int32) ["comparator" = "="]
+  '''
+  """
+  name = "rel_ssa.compare"
+
+  left = OperandDef(DataType)
+  right = OperandDef(DataType)
+  comparator = AttributeDef(StringAttr)
+
+  result = ResultDef(Boolean)
+
+  @staticmethod
+  @builder
+  def get(left: Operation, right: Operation,
+          comparator: StringAttr) -> 'Compare':
+    return Compare.build(operands=[left, right],
+                         attributes={"comparator": comparator},
+                         result_types=[Boolean])
+
+
+@irdl_op_definition
+class Yield(Expression):
+  """
+  Bridges the gap from expressions back to operators by yielding the result of
+  an expression to the encompassing operator.
+
+  Example:
+
+  '''
+  rel_ssa.yield(%0 : !rel_ssa.bool)
+  '''
+  """
+  name = "rel_ssa.yield"
+
+  ops = VarOperandDef(AnyAttr())
+
+  @staticmethod
+  @builder
+  def get(ops: list[Operation]) -> 'Yield':
+    return Yield.build(operands=[ops])
+
+
+@irdl_op_definition
+class Literal(Expression):
+  """
+  Creates a literal with value `value`.
+
+  Example:
+
+  '''
+  %0 : !rel_ssa.int32 = rel_ssa.literal() ["value" = 5 : i32]
+  '''
+  """
+  name = "rel_ssa.literal"
+
+  # TODO: change IntegerAttr s.t. it can have type !rel.int32
+  value = AttributeDef(AnyAttr())
+  result = ResultDef(DataType)
+
+  @staticmethod
+  @builder
+  def get(val: Attribute, res: DataType) -> 'Literal':
+    return Literal.build(attributes={"value": val}, result_types=[res])
+
+
+#===------------------------------------------------------------------------===#
+# Operators
+#===------------------------------------------------------------------------===#
+
+
+class Operator(Operation):
+  """
+  Interface class for all Operators, i.e., operations that work on bags.
+  """
+  ...
+
+
+@irdl_op_definition
+class PandasTable(Operator):
+  """
+  Defines a table with name `table_name`.
+
+  Example:
+
+  '''
+  %0 : rel_ssa.bag<[!rel_ssa.int32]> = rel_ssa.pandas_table() ["table_name" = "t"]
```

Comment:
ok, this makes sense to me.
I guess I have trouble understanding the Column op ..